### ISSUE TITLE
bug(COMPONENT): Angular mat-table UI not updating on datasource update - (signalr-azure service live update)

### ISSUE THREAD

**Issue by znjs** [angular/components#26269]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

We are using server-stream updates from the signalr-azure service to get live updates from the backend, and the WebSocket is blocked on the client-side(browser). 
We can observe that the data is being updated in debugging, but the mat-table UI is not updated without any mouse events.

### Reproduction

Steps to reproduce:
1. Create a signalr-service for live updates
2. Create a connection with signalr service and render the incoming data in a mat-table
3. Block the web-socket on client-side(browser), use event stream updates to receive data from signalr
4. Create a button to reload the data, onClick of the button expected to update the mat-table with recent data.

### Expected Behavior

mat-table should load the data onClick of the button

### Actual Behavior

mat-table `datasource` being updated but not reflected in the UI until any mouse event is performed (cursor movement)

### Environment

- Angular: 13.0.x
- CDK/Material: v 13.0.2
- Browser(s): Chrome Version 108.0.5359.98 (Official Build) (64-bit)
- Operating System (e.g. Windows, macOS, Ubuntu): Windows 10

**Comment by amysorto** [comment]
Closing this issue due to no recent activity, if this is still an issue please create a new issue with a reproduction.
